contract_api/consumers/service_event_consumer.py

```diff
@@ -345,7 +348,27 @@ def update_proto_stubs(org_id, service_id, proto_stubs):
             )
             service_media_repo.create_service_media(service_media=media_item)
 
-    def process_service_deployment(self, org_id, service_id, update_proto_stubs):
+    def upload_proto_file_from_hash_to_bucket(self, org_id, service_id, asset_hash):
+        temp_dir = tempfile.gettempdir()
+
+        base_path = os.path.join(temp_dir, str(uuid.uuid1()))
+        if not os.path.exists(base_path):
+            os.makedirs(base_path)
+
+        temp_download_path = os.path.join(base_path, 'proto.tar')
+        temp_extraction_path = os.path.join(base_path, 'proto')
+        temp_output_path = os.path.join(base_path, 'proto.tar.gz')
+
+        io_bytes = self._ipfs_util.read_bytesio_from_ipfs(asset_hash)
+        with open(temp_download_path, 'wb') as outfile:
+            outfile.write(io_bytes.getbuffer())
+
+        extract_zip_file(zip_file_path=temp_download_path, extracted_path=temp_extraction_path)
+        make_tarfile(source_dir=temp_extraction_path, output_filename=temp_output_path)
+        self._s3_util.push_file_to_s3(temp_output_path, ASSETS_COMPONENT_BUCKET_NAME,
```

Comment:
Can we confirm that the S# bucket is the same as the one used for all media assets. We want only one S3 bucket for all org and service data

Reply:
A Jira ticket is created to address this issue : https://singularitynet.atlassian.net/browse/SNET-814